src/utils/editor/source-search.js

```diff
@@ -201,13 +201,18 @@ function searchNext(ctx, rev, query, newQuery, modifiers) {
   cm.operation(function() {
     let state = getSearchState(cm, query, modifiers);
     const pos = getCursorPos(newQuery, rev, state);
+
+    if (!state.query) {
+      return;
+    }
+
     let cursor = getSearchCursor(cm, state.query, pos, modifiers);
 
     const location = rev
       ? { line: cm.lastLine(), ch: null }
       : { line: cm.firstLine(), ch: 0 };
 
-    if (!cursor.find(rev)) {
+    if (!cursor.find(rev) && state.query) {
```

Comment:
Yes. That's what I thought as well, but if we don't include this here, we'll get
<img width="752" alt="screen shot 2017-05-21 at 1 33 02 pm" src="https://cloud.githubusercontent.com/assets/4562118/26281469/07519296-3e2a-11e7-984f-0f250443849d.png">